### PR TITLE
G3 release

### PR DESCRIPTION
**Pull request by yiming-kang**
- Update README
- Clean up TODOs

### CODE REVIEW

File: CODE/explain_human_resps.py [hunk 1]
```diff
@@ -47,8 +47,6 @@ def parse_args(argv):
     parser.add_argument(
         '--enable_permutation_shap', action='store_true',
         help="Enable SHAP for permutation runs. Permutations will default to not calculating SHAP values\n(using this flag will make permutations training significantly longer.")
```
Comment (wsjung) [inline CODE/explain_human_resps.py]: If I understood correctly, you would like:
1. Default to calculating SHAP values for regular (non-permutation) runs
2. Default to not calculating SHAP values for permutation runs

Replacing `--disable_shap` with `--enable_permutation_shap` will not allow us to disable calculating SHAP values in regular runs. Perhaps default to not calculating SHAP values for both regular and permutation runs and change flag to `--enable_shap`:
```suggestion
        '--enable_shap', action='store_true',
        help="Enable calculation of SHAP values. Runs will default to not calculating SHAP values\n(using this flag will make training significantly longer.")
```

This change in flag name would also have to be reflected in parameter and argument names in lines #57 and #136.

File: CODE/explain_human_resps.py [hunk 2]
```diff
@@ -85,11 +83,6 @@ def run_tfpr(tf_feat_mtx_dict, nontf_feat_mtx, features, label_df_dict, output_d
         logger.info('==> Cross validating response prediction model <==')
         tfpr_explainer.cross_validate(permute=permutations)
 
-        # TODO: to be removed in release
-        if not disable_shap:
-            logger.info('==> Analyzing feature contributions <==')
-            tfpr_explainer.explain()
-
```
Comment (wsjung) [inline CODE/explain_human_resps.py]: Deleting this entire section will remove SHAP value calculations completely, regardless of permutation or not.
Two ways to go about this (depending on how flag design):

1. Default to calculating SHAP values for regular runs, but not for permutation runs.
```suggestion
        if (not permutation) or (permutation and enable_permutation_shap):
            logger.info('==> Analyzing feature contributions <==')
            tfpr_explainer.explain()
```
2. Default to not calculating SHAP values regardless of run type.

```suggestion
        if enable_shap:
            logger.info('==> Analyzing feature contributions <==')
            tfpr_explainer.explain()
```

File: CODE/explain_yeast_resps.py [hunk 3]
```diff
@@ -37,8 +37,6 @@ def parse_args(argv):
     parser.add_argument(
         '--model_config', default='MODEL_CONFIG/yeast_default_config.json',
         help='Json file for pretrained model hyperparameters.')
-    parser.add_argument(
-        '--disable_shap', action='store_true',)
```
Comment (wsjung) [inline CODE/explain_yeast_resps.py:41]: Are we not adding a flag for the yeast scripts for enabling SHAP value calculations like we did for human?

File: CODE/explain_yeast_resps.py [hunk 4]
```diff
@@ -96,11 +84,6 @@ def main(argv):
 
     logger.info('==> Cross validating response prediction model <==')
     tfpr_explainer.cross_validate()
-
-    # TODO: to be removed in release
-    if not args.disable_shap:
-        logger.info('==> Analyzing feature contributions <==')
-        tfpr_explainer.explain()
```
Comment (wsjung) [inline CODE/explain_yeast_resps.py]: Same issue as human script.

File: README.md [hunk 5]
```diff
@@ -1,63 +1,41 @@
...
-```
-$ python3 CODE/explain_yeast_resps.py \
-    --is_regressor \
-    -i freq \
-    -f histone_modifications chromatin_accessibility dna_sequence_nt_freq gene_expression gene_variation \
-    -x RESOURCES/h5_data/yeast_dna_cc_hm_atac_tss1000to500b_expr_var.h5 \
-    -y RESOURCES/Yeast_ZEV_IDEA/ZEV_15min_shrunkenData_deFreq.csv \
-    -o OUTPUT/Yeast_ZEV_DE_freq/tf_indep_feats/xgb/
-```
+For human genome, use `CODE/explain_human_resps.py` with the above set of arguments.
```
Comment (wsjung) [inline README.md:38]: We could add a description of the flag for enabling SHAP value calculation below.

**Review by wsjung — CHANGES_REQUESTED** [review]
Need a design decision for default behavior of calculating SHAP values.
As it stands, deletion of the existing `disable_shap` section removes SHAP value calculation completely.

File: CODE/explain_human_resps.py (hunk 1 above)
Comment (yiming-kang) [reply CODE/explain_human_resps.py]: My inclination would be disabling shap when `-permutations` is enabled as it doesn't make sense to analyze shap for random permutations.

File: CODE/explain_yeast_resps.py (hunk 3 above)
Comment (yiming-kang) [reply CODE/explain_yeast_resps.py:41]: I don't intend to offer that option. Disabling shap is only meant for running permutation efficiently.

File: CODE/explain_yeast_resps.py (hunk 4 above)
Comment (yiming-kang) [reply CODE/explain_yeast_resps.py]: Fixed.

File: README.md (hunk 5 above)
Comment (yiming-kang) [reply README.md:38]: Updated.

**Review by wsjung — APPROVED** [review]

**Pull request merged**